improver/utilities/temporal.py

```diff
@@ -427,60 +427,75 @@ def _fill_timezones(self, input_cube: Cube) -> None:
-    @staticmethod
-    def _get_time_bounds_offset(input_cube: Cube) -> Optional[ndarray]:
-        """Returns the generalised offset of bounds[0] and bounds[1] from points on the
-        time coord. Bound intervals must match as we have used MergeCubes, so only need
-        to access the first time point.
-        """
-        time_coord = input_cube.coord("time")
-        point = time_coord.points[0]
-        if time_coord.has_bounds():
-            bounds = time_coord.bounds[0]
-            return bounds - point
-        else:
-            return None
-
-    def check_input_cube_dims(self, input_cube: Cube, timezone_cube: Cube) -> None:
+        if input_cube.coord("time").bounds is not None:
+            bounds = []
+            # Index 0 = lower bound, index 1 = upper bound
+            for index in [0, 1]:
+                time_bounds = input_cube.coord("time").bounds[:, index]
+                time_bounds = time_bounds.reshape((1, 1, len(time_bounds))) * (
+                    1 - self.timezone_cube.data
+                )
+                bounds.append(time_bounds.sum(axis=-1))
+
+            self.time_bounds = np.stack(bounds, axis=-1)
+
+    def check_input_cube_dims(self, input_cube: Cube) -> None:
         """Ensures input cube has at least three dimensions: time, y, x. Promotes time
-        to be the inner-most dimension (dim=-1). Does the same for the timezone_cube
-        UTC_offset dimension.
+        to be the inner-most dimension (dim=-1).
 
         Raises:
             ValueError:
                 If the input cube does not have exactly the expected three coords.
                 If the spatial coords on input_cube and timezone_cube do not match.
         """
-        expected_coords = ["time"] + [input_cube.coord(axis=n).name() for n in "yx"]
+        time_coord_name = "time"
+        expected_coords = [time_coord_name] + [
+            input_cube.coord(axis=n).name() for n in "yx"
+        ]
         cube_coords = [coord.name() for coord in input_cube.coords(dim_coords=True)]
         if not all(
             [expected_coord in cube_coords for expected_coord in expected_coords]
         ):
-            raise ValueError(
-                f"Expected coords on input_cube: time, y, x ({expected_coords})."
-                f"Found {cube_coords}"
-            )
-        enforce_coordinate_ordering(input_cube, ["time"], anchor_start=False)
-        self.timezone_cube = timezone_cube.copy()
-        enforce_coordinate_ordering(
-            self.timezone_cube, ["UTC_offset"], anchor_start=False
-        )
+            try:
+                time_aux_dim = input_cube.coord_dims(time_coord_name)
+            except CoordinateNotFoundError as err:
+                raise CoordinateNotFoundError(
+                    f"Expected coords on input_cube: time, y, x ({expected_coords})."
+                    f"Found {cube_coords}"
+                ) from err
+
+            temporary_time_dimcoord = input_cube.coord(time_coord_name).copy()
```

Comment:
That's not quite what is going on here, it's the construction of a new time coordinate when iris can't make one because of non-monotonic bounds. I've significantly extended the doc string to explain this.